### DISCUSSION TITLE
What happens when you run out of recovery codes?

### DISCUSSION THREAD

**Discussion by CodeDmitry** [community/community#76225] [New to GitHub]
I just activated 2FA on my account and I am curious to know what happens if I run out of recovery codes. 

I tried testing this with a test account before activating 2FA on my main account, but that account ended up getting locked out due to sending too many 2FA requests in a short period of time. I only have 16 recovery codes, and based on experience from my alt account, it does not seem possible to get any more recovery codes after all of the 16 are exhausted.

Is this account now officially on a 17-lives system(3 if I lose the 2fa device, local recovery codes, and access to remote recovery codes) with no way out other than migrating all my projects to an another account that isn't forced to activate 2FA?

**Comment by Miguel-Sanz** [comment]
Hello, CodeDmitry

Thank you for reaching out with your inquiry regarding the recovery codes and the implications of running out of them after enabling 2FA on your GitHub account.

While it's true that GitHub provides a limited number of recovery codes (in your case, 16), and it's crucial to manage them carefully, there are additional steps you can take to ensure the security and accessibility of your account.

If you find yourself running low on recovery codes, consider the following:

Regenerating Recovery Codes:
GitHub allows you to regenerate a new set of recovery codes after you've used a certain number of them. This is to ensure that you always have a set of valid recovery codes available. You can typically find this option in your account security settings.

Contact GitHub Support:
If you exhaust all recovery codes and face difficulties, it's advisable to reach out to GitHub support directly. They can assist you in regaining access to your account securely. GitHub's support team is well-equipped to handle such situations and can guide you through the necessary steps.

Account Recovery Options:
GitHub may have additional account recovery options beyond recovery codes. These could include verifying your identity through email, phone number, or other means. Exploring these options can provide additional layers of security and access.

Remember, the goal of 2FA is to enhance the security of your account. It's important to balance this with measures to ensure you can regain access in case of unforeseen circumstances.

If you have specific concerns or encounter challenges, GitHub support is your best resource for personalized assistance.

I hope this information is helpful. If you have further questions or need additional guidance, please don't hesitate to ask.
